proxy/http2/Http2ConnectionState.cc

```diff
@@ -756,6 +773,15 @@ Http2ConnectionState::main_event_handler(int event, void *edata)
     return 0;
   }
 
+  case HTTP2_SESSION_EVENT_XMIT: {
+    send_data_frames_depends_on_priority();
+
+    SCOPED_MUTEX_LOCK(lock, mutex, this_ethread());
+    _scheduled = false;
```

Comment:
Huh, I'm merely glancing here, but it seems really strange that we'd need to hold a mutex just to set _scheduled to false? Could that not at least be done with just a CAS? Seeing this, also makes me wonder is if the send_data_frames_depends_on_priority() call should also be protected under the mutex?